### DISCUSSION TITLE
What plugins are available for rabbitmq to analyze or view unacked messages?

### DISCUSSION THREAD

**Discussion by ponponon** [rabbitmq/rabbitmq-server#12251] [Questions] [answered]
Requirement scenario: I want to see what these unacked messages are

![图片](https://github.com/user-attachments/assets/c2241a21-60f5-4fcd-9c0e-abe6b7fd3564)

But I can't see it now, because the get messages function doesn't allow you to view unacked messages.

![图片](https://github.com/user-attachments/assets/064decf0-8bf7-47cf-8751-5d293f17bf0f)

But I think it's a normal requirement, for example, when there is a consumption block, I want to know which messages are causing the blockage

**Comment by michaelklishin** [comment, marked as answer]
RabbitMQ does not provide metrics for individual messages, only totals per queue.

If you want to know what messages are not being acknowledged, you have to introduce such metric in your own applications.
